conans/client/rest/rest_client_v1.py

```diff
@@ -306,10 +307,13 @@ def _remove_conanfile_files(self, ref, files):
     @handle_return_deserializer()
     def remove_packages(self, ref, package_ids=None):
         """ Remove any packages specified by package_ids"""
-        self.check_credentials()
-        payload = {"package_ids": package_ids}
-        url = self.router.remove_packages(ref)
-        return self._post_json(url, payload)
+        pcks = self.search_packages(ref, query=None)
```

Comment:
I would only do the search if ``package_ids is None``, in other cases it should still raise.

Actually, I'd prefer capturing the raise, checking if package_ids is None, and re-raising if not, but I guess it is not easy to differentiate the 404 that the ``packages`` folder is missing or the 404 that the recipe is missing.